packages/modules/vehicles/smarteq/api.py

```diff
@@ -345,83 +735,457 @@ def get_status(self, vin: str) -> int:
+                        self.vhList = self.su.get_vin_ev_map()
+                    if self._vin in self.vhList:
+                        ev = self.vhList[self._vin]
+                        self.su.set_CarState(ev, self.ws_soc, self.ws_range, self.ws_ts)
+                    else:
+                        self.log.warning('ws_on_message: VIN not configured as smarteq vehicle: ' + self._vin)
+                # restore range if X flag is set for testing
+                if 'X' in self.logFilter:
+                    self.ws_range = self.ws_range_X
+                    self.ws_store['range'] = self.ws_range
+
+                # check time - if connect is more than TOKENS_REFRESH_THRESHOLD ago, stop connection
+                now = int(time.time())
+                if now > int(self.ws_connect_time) + TOKENS_REFRESH_THRESHOLD_WS:
+                    self._infoLog('m', "ws_on_message: close connection")
+                    ws.keep_running = False
+            else:
+                self.log.warning("ws_on_message: wrong VIN, continue listening")
+        elif messageType == 'assignedVehicles':
+            if self.assignedVehicles_cnt == 0:
+                self.assignedVehicles_cnt += 1
+                self.write_ws_message(messageType, dmsg)
+                self._vins = list(dmsg['assignedVehicles']['vins'])
+                for ivin in self._vins:
+                    self._infoLog('i', "ws_on_message: assignedVehicles: " + ivin)
+                    tn = self.ws_fname(ivin)
+                    if not os.path.exists(tn):
+                        # ws file for vin does not extst. create new
+                        self.init_ws_store(ivin)
+                        self.write_ws_store(ivin)
+        elif messageType == 'debugMessage':
+            if self.debugMessage_cnt == 0:
+                self.debugMessage_cnt += 1
+                self.write_ws_message(messageType, dmsg)
+        elif messageType == 'apptwinPendingCommandRequest':
+            if self.apptwinPendingCommandRequest_cnt == 0:
+                self.apptwinPendingCommandRequest_cnt += 1
+                self.write_ws_message(messageType, dmsg)
+        else:
+            self.write_ws_message(messageType, dmsg)
+
+    def ws_on_error(self, ws, error):
+        self.log.error("ws_on_error: " + str(error))
+        try:
+            ws.close()
+        except Exception as e:
+            self._infoLog('w', "ws_on_error: close exception: " + ", error=" + str(e))
+        self.ws_soc = str(-1)
+        self.ws_range = str(0.0)
+        self.ws_soc_ts = 'na'
+        self.write_ws_store(self.vin)
+        self.wsError = 1
+
+    def ws_on_close(self, ws, status_code, message):
+        self._infoLog('w', "ws_on_close, status_code=" + str(status_code) +
+                      ", message=" + str(message))
+        self.ws_soc = str(-1)
+        self.ws_range = str(0.0)
+        self.ws_soc_ts = 'na'
+        self.write_ws_store(self.vin)
```

Comment:
Wenn die Abfrage fehl schlägt, bitte eine Exception werfen und keine Standardwerte setzen. Wird die Abfrage dreimal mit einer Exception abgebrochen, startet das Backend bei SoC-geführtem Lademodus die Ladung, damit man nicht mit einem leeren Auto da steht, weil der Server nicht erreichbar war.

Reply:
Das passiert in dem Websocket thread und signalisiert der nächsten regulären Abfrage, dass die WebSocket-Schnittstelle keine Daten liefern konnte...